engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -3208,9 +3222,7 @@ namespace dmGraphics
                 params.m_MipMap, layer_count);
             CHECK_VK_ERROR(res);
 
-            uint32_t slice_size = texDataSize / layer_count;
-
-            VkBufferImageCopy vk_copy_regions[6];
+            VkBufferImageCopy* vk_copy_regions = new VkBufferImageCopy[layer_count];
```

Comment:
What is the max layer count?
If we can avoid dynamic mem allocations with little impact, I think we prefer that.